client/core/account.go

```diff
@@ -190,3 +191,26 @@ func (c *Core) AccountImport(pw []byte, acct Account) error {
 
 	return nil
 }
+
+// UpdateCert attempts to connect to a server using a new TLS certificate. If
+// the connection is successful, then the cert in the database is updated.
+func (c *Core) UpdateCert(host string, cert []byte) error {
+	accountInfo, err := c.db.Account(host)
+	if err != nil {
+		return err
+	}
+
+	accountInfo.Cert = cert
+
+	_, connected := c.connectAccount(accountInfo)
```

Comment:
If there's a dexConnection already in the map, it will be orphaned.  You can observe this if a failing reconnect loop is going for the old connection as it never stops even though the conn was replaced with a working one with a new TLS cert.

I believe `connectAccount` needs a check that stops any existing connection (prior to connecting the new one otherwise the frontend gets the notifications in the wrong order and it looks disconnected until you refresh the page):

```diff
diff --git a/client/core/core.go b/client/core/core.go
index 48b1002f..4dfcf333 100644
--- a/client/core/core.go
+++ b/client/core/core.go
@@ -4824,7 +4824,8 @@ func (c *Core) initialize() {
 // connectAccount makes a connection to the DEX for the given account. If a
 // non-nil dexConnection is returned, it was inserted into the conns map even if
 // the initial connection attempt failed (connected == false), and the connect
-// retry / keepalive loop is active.
+// retry / keepalive loop is active. If there was already a dexConnection, it is
+// first stopped.
 func (c *Core) connectAccount(acct *db.AccountInfo) (dc *dexConnection, connected bool) {
        if !acct.Paid && len(acct.FeeCoin) == 0 {
                // Register should have set this when creating the account that was
@@ -4839,6 +4840,14 @@ func (c *Core) connectAccount(acct *db.AccountInfo) (dc *dexConnection, connecte
                c.log.Errorf("skipping loading of %s due to address parse error: %v", host, err)
                return
        }
+
+       c.connMtx.RLock()
+       if dc := c.conns[host]; dc != nil {
+               dc.connMaster.Disconnect()
+               dc.acct.lock()
+       } // leave it in the map so it remains listed if connectDEX fails
+       c.connMtx.RUnlock()
+
        dc, err = c.connectDEX(acct)
        if dc == nil {
                c.log.Errorf("Cannot connect to DEX %s: %v", host, err)
```

Reply:
Should close the book feeds to I think

```go
	c.connMtx.RLock()
	if dc := c.conns[host]; dc != nil {
		dc.connMaster.Disconnect()
		dc.acct.lock()
		dc.booksMtx.Lock()
		for _, b := range dc.books {
			b.closeFeeds()
			if b.closeTimer != nil {
				b.closeTimer.Stop()
			}
		}
		dc.booksMtx.Unlock()
	} // leave it in the map so it remains listed if connectDEX fails
	c.connMtx.RUnlock()
```

Reply:
https://github.com/decred/dcrdex/commit/ed9284e99e17aef82f8af4d3ef7a5d0527bad5ed

Reply:
Nice, the changes work well.